src/worker.cpp

```diff
@@ -466,31 +469,71 @@ void Worker::setDeviceTreeAndJson()
                                         ec);
         if (ec)
         {
-            throw std::runtime_error(
-                "create_symlink system call failed with error" + ec.message());
+            if (ec.message().compare("File exists") ==
+                constants::STR_CMP_SUCCESS)
+            {
+                ec.clear();
+                logging::logMessage(
+                    "Sym link already exists, file [" +
+                    std::string(INVENTORY_JSON_SYM_LINK) +
+                    "] deleting and creating sym link again, target file: " +
+                    systemJson);
+
+                if (std::filesystem::remove(INVENTORY_JSON_SYM_LINK, ec))
+                {
+                    std::filesystem::create_symlink(
+                        systemJson, INVENTORY_JSON_SYM_LINK, ec);
+                    if (ec)
+                    {
+                        throw std::runtime_error(
+                            "create_symlink system call failed with error: " +
+                            ec.message());
+                    }
+                }
+                else
+                {
+                    logging::logMessage(
+                        "remove system call failed for file[" +
+                        std::string(INVENTORY_JSON_SYM_LINK) + "], error[" +
+                        ec.message() + "], continuing with existing sym link.");
+                }
+            }
+            else
+            {
+                throw std::runtime_error(
+                    "create_symlink system call failed with error: " +
```

Comment:
missing symlink source and target path

Reply:
Also the error message in line 504 and line 489 are same. While debugging it is not easy to figure out where we hit the issue. So let's have a description which helps us in debugging.